Core.ApplicationServices/SystemUsage/Migration/ItSystemUsageMigrationServiceAdapter.cs

```diff
@@ -0,0 +1,122 @@
+    {
+        private readonly IEntityIdentityResolver _identityResolver;
+        private readonly IItSystemUsageMigrationService _systemUsageMigrationService;
+        private readonly IAuthorizationContext _authorizationContext;
+        private readonly ITransactionManager _transactionManager;
+        private readonly IOrganizationService _organizationService;
+
+        public ItSystemUsageMigrationServiceAdapter(IEntityIdentityResolver identityResolver,
+            IItSystemUsageMigrationService systemUsageMigrationService,
+            IAuthorizationContext authorizationContext,
+            ITransactionManager transactionManager,
+            IOrganizationService organizationService)
+        {
+            _identityResolver = identityResolver;
+            _systemUsageMigrationService = systemUsageMigrationService;
+            _authorizationContext = authorizationContext;
+            _transactionManager = transactionManager;
+            _organizationService = organizationService;
+        }
+
+        public Result<ItSystemUsageMigration, OperationError> GetMigration(Guid usageUuid, Guid toSystemUuid)
+        {
+            return ResolveUsageAndSystemIds(usageUuid, toSystemUuid)
+                .Bind(result => _systemUsageMigrationService.GetSystemUsageMigration(result.usageId, result.systemId))
+                .Bind(migration => WithReadAccess(migration.SystemUsage)
+                    .Select(_ => migration)
+                );
+        }
+
+        public Result<ItSystemUsage, OperationError> ExecuteMigration(Guid usageUuid, Guid toSystemUuid)
+        {
+            var transaction = _transactionManager.Begin();
+
+            return ResolveUsageAndSystemIds(usageUuid, toSystemUuid)
+                .Bind(result => _systemUsageMigrationService.ExecuteSystemUsageMigration(result.usageId, result.systemId))
+                .Bind(WithModifyAccess)
+                .Match(
+                    usage =>
+                    {
+                        transaction.Commit();
+                        return Result<ItSystemUsage, OperationError>.Success(usage);
+                    },
+                    error =>
+                    {
+                        transaction.Rollback();
+                        return error;
+                    }
+                );
+        }
+
+        public Result<IEnumerable<ItSystem>, OperationError> GetUnusedItSystemsByOrganization(Guid organizationUuid,
+            int numberOfItSystems,
+            bool getPublicFromOtherOrganizations,
+            params IDomainQuery<ItSystem>[] conditions)
+        {
+            return _organizationService.GetOrganization(organizationUuid)
+                .Bind(organization => _systemUsageMigrationService.GetUnusedItSystemsByOrganizationQuery(organization.Id, numberOfItSystems, getPublicFromOtherOrganizations, conditions))
+                .Select<IEnumerable<ItSystem>>(x => x.ToList());
```

Comment:
if you just return the IQueryable this method will be simpler and you can leave it up to your controller to materialize the result with ToList